cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/PropertyUtils.java

```diff
@@ -0,0 +1,45 @@
+package cloud.foundry.cli.crosscutting.mapping;
+
+import cloud.foundry.cli.crosscutting.logging.Log;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Utils class for reading information from property files.
+ */
+public class PropertyUtils {
+
+    private static final Log log = Log.getLog(PropertyUtils.class);
+
+    /**
+     * Name of the property file, which contains the version of our application
+     */
+    private static final String VERSION_PROPERTIES = "version.properties";
+
+    /**
+     * The key of the version in the properties file
+     */
+    private static final String VERSION = "version";
+
+
+    /**
+     * Determines the API-Version from our application
+     * If it fails, the API Version will be marked as NOT_FOUND and the error will be logged.
+     *
+     * @return API-Version
+     */
+    public String determineApiVersion(ResourceProvider resourceProvider, Properties properties) {
```

Comment:
This should either become static, or you should use the singleton pattern. There's no need to create instances of this class.

Reply:
If I make this method static, I can not test it at all.

Reply:
That's a strong argument towards using the singleton pattern then, together with dependency injection (which you already have) and a proper interface. This'll also allow for testing then.

Reply:
I made the call static. A singleton is imo still wrong.